aidial_adapter_vertexai/chat/static_tools.py

```diff
@@ -27,21 +27,62 @@ def parse_gemini_tools(
     ) -> List[GeminiTool] | None: ...
 
 
+class DynamicRetrievalConfig(BaseModel):
+    mode: Literal["MODE_DYNAMIC", "MODE_UNSPECIFIED"] | None = None
+    dynamic_threshold: float | None = None
+
+    @classmethod
+    def root_validator(cls, values):
+        if (
```

Comment:
Why not delegate this validation to VertexAI?

Reply:
They have bad handling of `0 < dynamic_threshold < 1`, which will confuse user

On `dynamic_threshold = 10` library sends request, that results into 400 with just `invalid argument`
Bad UX